### ISSUE TITLE
Spurious characters appearing in comments when saving file

### ISSUE THREAD

**Issue by grayswandyr** [ocamllabs/vscode-ocaml-platform#129]
Hi, 
using v0.4.0 and the last version of ocaml-lsp : in *one* file I have, spurious non-printable characters appear in comments each time I save. The file path is only made of ASCII symbols, as well as the file content itself.

![Untitled](https://user-images.githubusercontent.com/8328276/78882957-23d41100-7a59-11ea-9b12-8a21b439e7b1.png)

**Comment by imbsky** [comment]
This is probably a issue that should be reported to the repository on the OCaml LSP side. CC: @rgrinberg @cannorin @giltho

**Comment by giltho** [comment]
Hi! 
That's definitely a curious one 🤔 
Do you have somewhere I could reproduce this ?

Otherwise, can you set the setting `ocaml.trace.server` to `verbose`
(by adding
```json
"ocaml.trace.server": "verbose"
```
in your vscode's settings.json)
Then, restart your vscode, open the file, reproduce the bug for a while, and then send us a file where you copy-paste the logs you obtain in  `output`->`OCaml language server` ?

My intuition is that this bug comes from a different extension. Saving should not trigger anything from ocaml-lsp.

**Comment by imbsky** [comment]
I think this is a issue with OCamlFormat.

**Comment by imbsky** [comment]
@grayswandyr Maybe you are using OCamlFormat, right? Try disabling save on format mode before saving.

**Comment by grayswandyr** [comment]
Thanks for your answers. I am also using OcamlFormat so it may come from here too. I added the "trace" option as requested, I'll update you if I can reproduce the bug (as I wrote initially, the bug happened on only one OCaml file in my project, and I can't reproduce it currently). Also, I has an old option that I just commented::
```
    // "[ocaml]": {
    //     "editor.defaultFormatter": "freebroccolo.reasonml"
    // },
```
I'll update you if the bug shows again.

**Comment by imbsky** [comment]
This seems to be a issue on the LSP side. Rudi already knows about it and will fix it soon.

**Comment by yannl35133** [comment]
This issue still has the tag "needs investigation", and I have run into this bug, so here is my version of it :
- First, the character  � ("\uFFFD") mysteriously appears in the file. I didn't manage to reproduce that part, it happened as I wrote an unbound constructor (e.g. `Nbb` when only `Nb` is defined). I saw VSCode show me the OCaml Language Server output to print an error, I don't have it anymore.
- Then, when you want to delete it, and save, it reappears, and multiplies. I can reproduce it, simply paste it in a blank `.ml` file, save, delete, save again.

If you cannot reproduce it, I can give you my output, but the  `"ocaml.trace.server": "verbose"` doesn't seem to be recognized, and it doesn't look useful as it is right now
